vulnerable_people_form/integrations/persistence.py

```diff
@@ -0,0 +1,199 @@
+import boto3
+
+from flask import current_app
+
+
+def get_rds_data_client(app=current_app):
+    return boto3.client(
+        "rds-data", endpoint_url=app.config.get("LOCAL_AWS_ENDPOINT_URL")
+    )
+
+
+def get_rds_client(app=current_app):
+    return boto3.client("rds", endpoint_url=app.config.get("LOCAL_AWS_ENDPOINT_URL"))
+
+
+def get_secretsmanager_client(app=current_app):
+    return boto3.client(
+        "secretsmanager", endpoint_url=app.config.get("LOCAL_AWS_ENDPOINT_URL")
+    )
+
+
+def _find_database_arn(app):
+    rds = get_rds_client(app=app)
+    clusters = rds.describe_db_clusters()["DBClusters"]
+    coronavirus_rds = [
+        x
+        for x in clusters
+        if x["DBClusterIdentifier"].startswith(app.config["DATABASE_CLUSTER_PREFIX"])
+    ][0]
+    return coronavirus_rds["DBClusterArn"]
+
+
+def _find_database_secret_arn(app):
+    secrets_manager = get_secretsmanager_client(app=app)
+    return secrets_manager.list_secrets(
+        Filters=[{"Key": "tag-value", "Values": app.config["DATABASE_SECRET_TAGS"]}]
+    )["SecretList"][0]["ARN"]
+
+
+def init_app(app):
+    if app.config.get("AWS_RDS_DATABASE_ARN_OVERRIDE"):
+        app.config["AWS_RDS_DATABASE_ARN"] = app.config["AWS_RDS_DATABASE_ARN_OVERRIDE"]
+    else:
+        app.config["AWS_RDS_DATABASE_ARN"] = _find_database_arn(app)
+
+    if app.config.get("AWS_RDS_SECRET_ARN_OVERRIDE"):
+        app.config["AWS_RDS_SECRET_ARN"] = app.config["AWS_RDS_SECRET_ARN_OVERRIDE"]
+    else:
+        app.config["AWS_RDS_SECRET_ARN"] = _find_database_secret_arn(app)
+
+
+def generate_string_parameter(name, value):
+    return {
+        "name": name,
+        "value": {"isNull": True} if value is None else {"stringValue": value,},
+    }
+
+
+def generate_int_parameter(name, value):
+    return {
+        "name": name,
+        "value": {"doubleValue": value,},
+    }
+
+
+def generate_bigint_parameter(name, value):
+    return {
+        "name": name,
+        "value": {"longValue": value,},
+    }
+
+
+def generate_date_parameter(name, value):
+    return {
+        "name": name,
+        "typeHint": "DATE",
+        "value": {"stringValue": "{year}-{month:0>2}-{day:0>2}".format(**value),},
+    }
+
+
+def execute_sql(sql, parameters):
+    return get_rds_data_client().execute_statement(
+        sql=sql,
+        parameters=parameters,
+        database=current_app.config["AWS_RDS_DATABASE_NAME"],
```

Comment:
i don't think the DB name is required and its a value that i don't think has been provided. The connection can work of the secret and the resourcearn

Reply:
Good point! Updated 👍